sympy/integrals/integrals.py

```diff
@@ -1242,6 +1242,144 @@ def _sage_(self):
                                     hold=True)
         return f
 
+    def cauchy_principal_value(self, sym, lower_limit=-oo, upper_limit=oo):
+        """
+        Compute the Cauchy Principal Value of the definite integral of a real function in the given interval
+        on the real axis.
+        In mathematics, the Cauchy principal value, is a method for assigning values to certain improper
+        integrals which would otherwise be undefined.
+        The Cauchy principal value of a finite integral of a function f about a point c with a <= c <= b is given by
+                b               c-r                       b
+              /                  /                      /
+             /                  /                      /
+             |                  |                      |
+             |                  |                      |
+        p.v. | f(x)dx  = lim    | f(x)dx   +   lim     |
+             |          r -> 0+ |             r -> 0+  |
+             |                  |                      |
+             /                  /                     /
+            /                  /                     /
+          a                 a                     c+r
+
+        Examples
+        ========
+
+        >>> from sympy import Dummy, symbols, integrate, limit, oo
+        >>> from sympy.integrals.integrals import Integral
+        >>> from sympy.calculus.singularities import singularities
+        >>> x = symbols('x', real = True)
+
+        >>> g = x + 1
+        >>> Integral(g).cauchy_principal_value(x)
+        oo
+
+        >>> f = 1/(x**3)
+        >>> singularities(f,x)
+        {0}
+
+        >>> Integral(f).cauchy_principal_value(x)
+        0
+
+        >>> Integral(f).cauchy_principal_value(x, -oo, 0)
```

Comment:
This should be `Integral(f, (x, -oo, 0)).principal_value()`. The limits are assigned to `Integral`.

Reply:
@jksuom  I have made these changes in the latest commit.